dbt/adapters/athena/impl.py

```diff
@@ -385,28 +386,38 @@ def _get_one_catalog(
     ) -> agate.Table:
         data_catalog = self._get_data_catalog(information_schema.path.database)
         catalog_id = get_catalog_id(data_catalog)
-        conn = self.connections.get_thread_connection()
-        client = conn.handle
-        with boto3_client_lock:
-            glue_client = client.session.client("glue", region_name=client.region_name, config=get_boto3_config())
-
-        catalog = []
-        paginator = glue_client.get_paginator("get_tables")
-        for schema, relations in schemas.items():
-            kwargs = {
-                "DatabaseName": schema,
-                "MaxResults": 100,
-            }
-            # If the catalog is `awsdatacatalog` we don't need to pass CatalogId as boto3 infers it from the account Id.
-            if catalog_id:
-                kwargs["CatalogId"] = catalog_id
 
-            for page in paginator.paginate(**kwargs):
-                for table in page["TableList"]:
-                    if table["Name"] in relations:
-                        catalog.extend(self._get_one_table_for_catalog(table, information_schema.path.database))
+        if data_catalog["Type"] == "GLUE":
```

Comment:
great work done, the code looks good to me. However I have a question. According to the [AWS doc](https://docs.aws.amazon.com/athena/latest/APIReference/API_DataCatalog.html) there are 3 possible options for catalog type: `GLUE`, `HIVE` and `LAMBDA`. You have implemented `LAMBDA` case I guess. If it's so, let's add kind of Enum for CatalogTypes and at least log warning or raise a NotImplemented error for HIVE catalog type

Reply:
thanks, I've added some minor comments regarding modules structure